src/test/datascience/notebook/interrupRestart.vscode.test.ts

```diff
@@ -204,4 +207,90 @@ suite('DataScience - VSCode Notebook - Restart/Interrupt/Cancel/Errors (slow)',
         // Stop the cell (cleaner way to tear down this test, else VS Code can hang due to the fact that we delete/close notebooks & rest of the code is trying to access it).
         vscEditor.kernel!.cancelAllCellsExecution(vscEditor.document);
     });
+    test('Interrupt and running cells again should only run the necessary cells', async function () {
+        // Interrupts on windows doesn't work well, not as well as on Unix.
+        // This is how Python works, hence this test is better tested on Unix OS.
+        // No need to test remote as this is a test of status (fewer slower tests is better).
+        if (getOSType() === OSType.Windows || IS_REMOTE_NATIVE_TEST) {
+            return this.skip();
+        }
+
+        /*
+        Cells 1, 2, 3.
+        Ensure cell 2 is a long running cell
+        Run all cells, interrupt execution when running cell 2.
+        Confirm 1 is a success & 2 has failed (as a result of interrupt)
```

Comment:
This may be flakey. Interrupt has always been hard to consistently interrupt.

Reply:
Hopefully not, i've limited the tests to just Unix.

Reply:
FYI - I believe I resolved the flaky ness of the interrupts